crates/bevy_pbr/src/render/pbr_types.wgsl

```diff
@@ -6,6 +6,12 @@ struct StandardMaterial {
     perceptual_roughness: f32,
     metallic: f32,
     reflectance: f32,
+    diffuse_transmission: f32,
+    transmission: f32,
+    thickness: f32,
+    ior: f32,
+    attenuation_distance: f32,
+    attenuation_color: vec4<f32>,
```

Comment:
I'm a bit concerned about how large this makes PbrInput. I think with these added it's around 1.5KB. Maybe there should be a define that only includes these when a material has something set saying it's going to use transmission.

Reply:
Is it because of alignment? These are 5*4+16 bytes, so 36 bytes optimally. With the worst alignment I can think of (each `f32` taking the space of one full `vec4<f32>`, which I don't think would happen) it would take up 96 bytes, which would still be very far from bringing it up to 1.5KB

Reply:
I'm referring to the total size of PbrInput. But I just realized that I calculated bits not bytes. I think it's closer to 188 bytes. Still quite large imo, though not nearly as ridiculous as 1.5KB

struct PbrInput {
    material: StandardMaterial {
        base_color: vec4<f32>,
        emissive: vec4<f32>,
        perceptual_roughness: f32,
        metallic: f32,
        reflectance: f32,
        diffuse_transmission: f32,
        transmission: f32,
        thickness: f32,
        ior: f32,
        attenuation_distance: f32,
        attenuation_color: vec4<f32>,
        flags: u32,
        alpha_cutoff: f32,
        parallax_depth_scale: f32,
        max_parallax_layer_count: f32,
        max_relief_mapping_search_steps: u32,
    },
    occlusion: vec3<f32>,
    frag_coord: vec4<f32>,
    world_position: vec4<f32>,
    world_normal: vec3<f32>,
    N: vec3<f32>,
    V: vec3<f32>,
    is_orthographic: bool,
    flags: u32,
};

Reply:
Yeah, it's a bit large, I wonder if the compiler is smart enough to not copy it with the function calls, and instead pass it by reference implicitly?

I was thinking maybe we could bring those down to `f16` since we don't really need a lot of precision for these values, but it looks like `f16` is an extension in WebGPU, (`"shader-f16"`) and not guaranteed to be available everywhere. 😕 

I am going to add a pipeline key / shader def for diffuse transmission, as suggested by @JMS55. We can reuse it for `diffuse_transmission` at least, and perhaps add a separate one for specular transmission & friends? Is the overhead of compiling, keeping independent shader programs in memory, and switching to them for the cases where we use transmission/diffuse transmission low enough to compensate the potential performance gains of keeping the structure smaller?

Reply:
We need the shader def for other perf and compatibility reasons, so I wouldn't worry about it.